### DISCUSSION TITLE
Unable to Obtain Text Output from Inference using ONNX Model

### DISCUSSION THREAD

**Discussion by fardinkhanz** [mindee/doctr#1517] [Q&A]
### Description
I have successfully exported an both the detection(db_resnet50) and recognisation(crnn_vgg16_bn) model of Doctr to ONNX format using the provided utility, but I am facing difficulty in obtaining text output from the inference process. Instead of receiving text outputs, I'm getting numerical matrices.

### Steps to Reproduce
1. Exported the detection and recognisation model of Doctr using the provided export utility.
2. Prepared input images for inference.
3. Ran inference on the input images using the exported ONNX model.
4. Received numerical matrices as output instead of text.

### Expected Behavior
I expected to receive text outputs from the inference process, representing the recognized text in the input images.

- In the below code I am just calling the recognisation model

### Example Code
```python
import numpy as np
import onnxruntime
import cv2

# Load the exported ONNX model
model_path = "path_to_your_model.onnx"
ort_session = onnxruntime.InferenceSession(model_path)

# Define the input shape expected by the model
input_shape = (32, 128, 3)

# Function to preprocess image for inference
def preprocess_image(image_path, input_shape):
    # Load and resize the image
    image = cv2.imread(image_path)
    image = cv2.resize(image, (input_shape[1], input_shape[0]))
    # Convert image to float32 and normalize
    image = image.astype(np.float32) / 255.0
    # Convert image to NHWC format
    image = np.expand_dims(image, axis=0)
    return image

# Function to perform inference on the image
def infer_on_image(image_path, ort_session):
    # Preprocess the image
    input_data = preprocess_image(image_path, input_shape)
    # Perform inference
    ort_outs = ort_session.run(None, {"input": input_data})
    return ort_outs

# Path to the image for inference
image_path = "path_to_your_image.jpg"

# Perform inference on the image
ort_outs = infer_on_image(image_path, ort_session)

# Process the output if necessary
print("Inference result:", ort_outs)

**Comment by felixdittrich92** [comment]
Hi @fardinkhanz :wave:,
Correct what you get is the `logits` vector the models does not include postprocessing (because postprocessing contains non convertable steps) :)

So you would need to write your own postprocessor with numpy
As reference:
https://github.com/mindee/doctr/blob/dddc2df561bfeabcb385db18b2ddea33349411a9/doctr/models/recognition/parseq/pytorch.py#L385
